backend/src/core/ban.rs

```diff
@@ -0,0 +1,208 @@
+use std::sync::Arc;
+
+use crate::util::{
+    madness::Madness,
+    types::{Ban, Character, Entity},
+};
+
+pub struct BanService {
+    db: Arc<crate::DB>,
+}
+
+impl BanService {
+    pub fn new(database: Arc<crate::DB>) -> BanService {
+        BanService { db: database }
+    }
+
+    pub async fn character_bans(&self, character_id: i64) -> Result<Option<Vec<Ban>>, Madness> {
+        if let Some(bans) = self.active_bans(character_id, "Character").await? {
+            Ok(Some(bans))
+        } else {
+            if let Some(character) = sqlx::query!(
+                "SELECT corporation_id FROM `character` WHERE id=?",
+                character_id
+            )
+            .fetch_optional(self.db.as_ref())
+            .await?
+            {
+                if let Some(corporation_id) = character.corporation_id {
+                    if let Some(bans) = self.corporation_bans(corporation_id).await? {
+                        return Ok(Some(bans));
+                    }
+                }
+            }
+            Ok(None)
+        }
+    }
+
+    pub async fn corporation_bans(&self, corporation_id: i64) -> Result<Option<Vec<Ban>>, Madness> {
+        if let Some(bans) = self.active_bans(corporation_id, "Corporation").await? {
+            Ok(Some(bans))
+        } else {
+            if let Some(corporation) = sqlx::query!(
+                "SELECT alliance_id as `alliance_id?` FROM corporation WHERE id=?",
+                corporation_id
+            )
+            .fetch_optional(self.db.as_ref())
+            .await?
+            {
+                if let Some(alliance_id) = corporation.alliance_id {
+                    if let Some(bans) = self.aliance_bans(alliance_id).await? {
+                        return Ok(Some(bans));
+                    }
+                }
+            }
+
+            Ok(None)
+        }
+    }
+
+    pub async fn aliance_bans(&self, alliance_id: i64) -> Result<Option<Vec<Ban>>, Madness> {
+        if let Some(bans) = self.active_bans(alliance_id, "Alliance").await? {
+            Ok(Some(bans))
+        } else {
+            Ok(None)
+        }
+    }
+
+    pub async fn active_bans(
+        &self,
+        entity_id: i64,
+        entity_type: &str,
+    ) -> Result<Option<Vec<Ban>>, Madness> {
+        let now: i64 = chrono::Utc::now().timestamp();
+
+        let rows = sqlx::query!(
+            "SELECT 
+                ban.id,
+                entity_id,
+                entity_name,
+                entity_type,
+                issued_at,
+                public_reason,
+                reason,
+                revoked_at,
+                issuer.id AS `issued_by_id`,
+                issuer.name AS `issued_by_name`
+            FROM
+                ban
+            JOIN
+                `character` as issuer ON issued_by=issuer.id
+            WHERE
+                entity_id=? AND entity_type=? AND revoked_at IS NULL OR revoked_at > ?",
```

Comment:
```
entity_id=? AND entity_type=? AND (revoked_at IS NULL OR revoked_at > ?)",
``` 
 The way you have it now means that if there are any valid bans everyone is banned, this fixed the issue in my dev space.

Reply:
Confirming your fix works for the character and corporation-level bans. I am unable to test alliance-level bans as I do not have a character that belongs to an alliance. Push incoming.